scraper/doctolib/doctolib_center_scrap.py

```diff
@@ -30,35 +31,40 @@
 
 DOCTOLIB_DOMAINS = DOCTOLIB_CONF.get("recognized_urls", [])
 
-
 DOCTOLIB_WEIRD_DEPARTEMENTS = SCRAPER_CONF.get("dep_conversion", {})
 
-
 logger = get_logger()
 booking_requests = {}
 
 
+def run_departement_scrap(departement: str):
+    logger.info(f"[Doctolib centers] Parsing pages of departement {departement} through department SEO link")
+    centers_departements = parse_pages_departement(departement)
+    if centers_departements == 0:
+        raise Exception("No Value found for department {}, crashing")
+    return centers_departements
+
+
 def parse_doctolib_centers(page_limit=None) -> List[dict]:
     centers = []
     unique_center_urls = []
 
-    for departement in get_departements():
-        logger.info(f"[Doctolib centers] Parsing pages of departement {departement} through department SEO link")
-        centers_departements = parse_pages_departement(departement)
-        if centers_departements == 0:
-            raise Exception("No Value found for department {}, crashing")
-        centers += centers_departements
+    with multiprocessing.Pool(50) as pool:
+        center_lists = pool.imap_unordered(run_departement_scrap, get_departements())
+        centers = []
 
-    centers = list(filter(is_vaccination_center, centers))  # Filter vaccination centers
-    centers = list(map(center_reducer, centers))  # Remove fields irrelevant to the front
+        for center_list in center_lists:
+            centers.extend(center_list)
+        centers = list(filter(is_vaccination_center, centers))  # Filter vaccination centers
+        centers = list(map(center_reducer, centers))  # Remove fields irrelevant to the front
 
-    for item in list(centers):
-        if item.get("rdv_site_web") in unique_center_urls:
-            centers.remove(item)
-            continue
-        unique_center_urls.append(item.get("rdv_site_web"))
+        for item in list(centers):
```

Comment:
Tu ne peux pas juste faire un filtre ? Je ne comprends pas du tout l'intention de ce que tu veux faire

`[centre for centre in centres if centre.get("rdv_site_web") not in unique_center_urls]` non ?

Reply:
Pas très lisible en une ligne